tests/test_ophys/test_tools_roiextractors.py

```diff
@@ -1330,6 +1344,80 @@ def test_add_two_photon_series_roundtrip(self):
             assert self.imaging_plane_name in imaging_planes_in_file
             assert len(imaging_planes_in_file) == 1
 
+    def test_add_invalid_photon_series_type(self):
+        """Test error is raised when adding photon series with invalid 'photon_series_type'."""
+        with self.assertRaisesWith(
+            AssertionError,
+            "'photon_series_type' must be either 'OnePhotonSeries' or 'TwoPhotonSeries'.",
+        ):
+            add_photon_series(
+                imaging=self.imaging_extractor,
+                nwbfile=self.nwbfile,
+                metadata=self.two_photon_series_metadata,
+                photon_series_type="invalid",
+            )
+
+    def test_add_photon_series_inconclusive_metadata(self):
+        """Test error is raised when `photon_series_type` specifies 'TwoPhotonSeries' but metadata contains 'OnePhotonSeries'."""
+        with self.assertRaisesWith(
+            AssertionError,
+            "Received metadata for 'OnePhotonSeries' but `photon_series_type` was not explicitly specified.",
+        ):
+            add_photon_series(
+                imaging=self.imaging_extractor,
+                nwbfile=self.nwbfile,
+                metadata=self.one_photon_series_metadata,
+            )
+
+    def test_add_one_photon_series(self):
+        """Test adding one photon series with metadata."""
+
+        metadata = deepcopy(self.one_photon_series_metadata)
+        one_photon_series_metadata = metadata["Ophys"]["OnePhotonSeries"][0]
+        one_photon_series_metadata.update(
+            pmt_gain=60.0,
+            binning=2,
+            power=500.0,
+        )
+        add_photon_series(
+            imaging=self.imaging_extractor,
+            nwbfile=self.nwbfile,
+            metadata=metadata,
+            photon_series_type="OnePhotonSeries",
+        )
+        self.assertIn(self.one_photon_series_name, self.nwbfile.acquisition)
+        one_photon_series = self.nwbfile.acquisition[self.one_photon_series_name]
+        self.assertIsInstance(one_photon_series, OnePhotonSeries)
+        self.assertEqual(one_photon_series.pmt_gain, 60.0)
+        self.assertEqual(one_photon_series.binning, 2)
+        self.assertEqual(one_photon_series.power, 500.0)
+        self.assertEqual(one_photon_series.unit, "n.a.")
+
+    def test_add_one_photon_series_roundtrip(self):
+        add_photon_series(
+            imaging=self.imaging_extractor,
+            nwbfile=self.nwbfile,
```

Comment:
[This](https://github.com/catalystneuro/neuroconv/actions/runs/5144962618/jobs/9261952899?pr=462) is failing because of a bug that was already [fixed](https://github.com/NeurodataWithoutBorders/pynwb/pull/1660) in `pynwb 2.3.2`, but minimal installation requires `pynwb>=1.4.0,<2.3.2`

Reply:
OK, let me know how extending the PyNWB version goes; don't remember why we disabled that on 3.11 but it was probably a random issue that they were notified about and hopefully it's been fixed

Reply:
It was maybe because of `numpy`?
https://github.com/catalystneuro/neuroconv/blob/e7d4904bb07773d3b23649720ee38b319b13f4e0/requirements-minimal.txt#L1

pynwb 2.3.2 would install numpy==1.24.2
[numpy==1.24.2; python_version >= "3.8"](https://github.com/NeurodataWithoutBorders/pynwb/blob/a6fa9170f2ad6942d74eaf44c22cedf0b80f0fcd/requirements.txt#LL4C1-L4C39
)
Which results in [failing tests](https://github.com/catalystneuro/neuroconv/actions/runs/5187152583/jobs/9349176761) for the voxel/pixel masks from `assert_array_equal`
```
E           ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (10,) + inhomogeneous part.
````
I tried to fix in https://github.com/catalystneuro/neuroconv/pull/462/commits/9d5a758ef8296825f7a7c43bed89feb89e7ab15c with adding:

```python
def assert_masks_equal(mask, expected_mask):
    # check the shape
    assert len(mask) == len(expected_mask)

    # compare each sub-array individually
    for i in range(len(mask)):
        assert_array_equal(mask[i], expected_mask[i])
 ```